File: packages/core/lib/src/branding.ts
/**
 * Branding primitives — single source of truth for the lamalibre / lamaste
 * namespace split. Ecosystem-level surfaces use `ORG` (lamalibre); product-level
 * surfaces nest under `ORG.PROJECT` (lamalibre.lamaste).
 *
 * `LAMALIBRE_ORG` / `LAMALIBRE_PROJECT` env vars override defaults at runtime.
 */

import { homedir } from 'node:os';
import path from 'node:path';

export const ORG = process.env.LAMALIBRE_ORG ?? 'lamalibre';
export const PROJECT = process.env.LAMALIBRE_PROJECT ?? 'lamaste';

// macOS reverse-DNS (dot-separated, hierarchical)
export const ecosystemBundleId = (service: string): string => `com.${ORG}.${service}`;
export const productBundleId = (service: string): string => `com.${ORG}.${PROJECT}.${service}`;

// systemd / hyphen-separated unit names
export const ecosystemUnit = (service: string): string => `${ORG}-${service}`;
export const productUnit = (service: string): string => `${ORG}-${PROJECT}-${service}`;

// User filesystem roots
export const userEcosystemRoot = (): string => path.join(homedir(), `.${ORG}`);
export const userProductRoot = (): string => path.join(homedir(), `.${ORG}`, PROJECT);

// System filesystem roots
export const etcEcosystemRoot = (): string => `/etc/${ORG}`;
export const etcProductRoot = (): string => `/etc/${ORG}/${PROJECT}`;

File: packages/clients/desktop/src-tauri/src/branding.rs
//! Branding primitives — single source of truth for the lamalibre / lamaste
//! namespace split. Mirrors `@lamalibre/lamaste`'s `branding.ts`.
//!
//! Ecosystem-level surfaces (cloud creds, storage, local plugin host, deep-link
//! scheme) use `ORG`. Product-level surfaces (lamaste PKI, panel, agent, daemons)
//! nest under `ORG.PROJECT`.

#![allow(dead_code)]

use std::path::PathBuf;

pub const ORG: &str = "lamalibre";
pub const PROJECT: &str = "lamaste";

pub fn ecosystem_bundle_id(service: &str) -> String {
    format!("com.{}.{}", ORG, service)
}

pub fn product_bundle_id(service: &str) -> String {
    format!("com.{}.{}.{}", ORG, PROJECT, service)
}

pub fn ecosystem_unit(service: &str) -> String {
    format!("{}-{}", ORG, service)
}

pub fn product_unit(service: &str) -> String {
    format!("{}-{}-{}", ORG, PROJECT, service)
}

// ---------------------------------------------------------------------------
// Filesystem roots — mirror userEcosystemRoot/userProductRoot/etc* in branding.ts
// ---------------------------------------------------------------------------

/// User ecosystem root: `~/.${ORG}/`
pub fn user_ecosystem_root() -> Option<PathBuf> {
    dirs::home_dir().map(|h| h.join(format!(".{}", ORG)))
}

/// User product root: `~/.${ORG}/${PROJECT}/`
pub fn user_product_root() -> Option<PathBuf> {
    user_ecosystem_root().map(|p| p.join(PROJECT))
}

/// System ecosystem root: `/etc/${ORG}/`
pub fn etc_ecosystem_root() -> PathBuf {
    PathBuf::from(format!("/etc/{}", ORG))
}

/// System product root: `/etc/${ORG}/${PROJECT}/`
pub fn etc_product_root() -> PathBuf {
    etc_ecosystem_root().join(PROJECT)
}
